### PR TITLE
Create post

### LINKED ISSUE #6: Not able to create any blog post
Thank you for this project. I can register and sign in. Can see my profile (although not similar to what I see on the gif image). I can also see my information in the database. But I am not able to create blog post. I was expecting something like "Create" in the page after I signed in.

### PR DESCRIPTION
**Pull request by zablon-oigo**
# feat: Implement post creation functionality

- Implemented logic view for creating posts
- Added forms.py to manage the post-creation form
- Created a template create_post.html for the post-creation view
- Included link in base.html for easy navigation to post creation
- Defined a path in urls.py to handle post-creation requests
createpost

### CODE REVIEW

File: blog/views.py [hunk 1]
```diff
@@ -10,3 +11,18 @@ def home(request):
...
+@login_required
+def create_post(request):
+    if request.method == 'POST':
+        form=PostForm(request.POST)
+        if form.is_valid():
+            user=form.save(commit=False)
+            user.author=request.user
+            user.save()
+            messages.success(request, 'The post has been created successfully.')
+            return redirect('home')
```
Comment (Akash1362000) [inline blog/views.py]: The return redirect should be `blog-home`

File: blog/views.py [hunk 2]
```diff
@@ -10,3 +11,18 @@ def home(request):
...
+    else:
+        messages.error(request, 'Please correct the following errors:')
```
Comment (Akash1362000) [inline blog/views.py]: The error message `Please correct the following errors` should be shown to the user only when there are any errors

Error

**Review by Akash1362000 — CHANGES_REQUESTED** [review]
Before committing your code, please run the below three commands:

```
pip install -r requirements.dev.txt
pre-commit install
pre-commit run --all-files
```

File: blog/views.py (hunk 2 above)
Comment (zablon-oigo) [reply blog/views.py]: Great, lemme work on it

Comment (zablon-oigo) [reply blog/views.py]: Capture2
Capture3

**Review by Akash1362000 — APPROVED** [review]
Looks good!

**Pull request merged**